src/main/java/org/zalando/nakadi/service/SchemaService.java

```diff
@@ -35,4 +45,21 @@ public SchemaService(final SchemaRepository schemaRepository,
                         () -> schemaRepository.getSchemasCount(name)));
     }
 
+    public Result<EventTypeSchema> getSchemaVersion(final String name, final String version) {
+        final Matcher versionMatcher = VERSION_PATTERN.matcher(version);
```

Comment:
I would argue that data validation is responsibility of the layer above, in the controller. You could use some `@Pattern` validation on the controller directly. It would look like:

```java
public ResponseEntity<?> getSchemaVersion(@PathVariable("name") @Pattern("...") @Valid final String name
```